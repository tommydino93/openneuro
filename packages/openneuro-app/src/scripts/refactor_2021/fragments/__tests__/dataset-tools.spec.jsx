import React from 'react'
import { shallow } from 'enzyme'
import DatasetTools from '../dataset-tools.jsx'

describe('DatasetTools component', () => {
  it('renders tools with common props', () => {
    const wrapper = shallow(
      <DatasetTools
        dataset={{
          id: 'ds001',
          public: false,
          following: true,
          starred: false,
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders tools for public datasets', () => {
    const wrapper = shallow(
      <DatasetTools
        dataset={{
          id: 'ds001',
          public: true,
          following: true,
          starred: false,
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders tools for public snapshots', () => {
    const wrapper = shallow(
      <DatasetTools
        dataset={{
          id: 'ds001',
          public: true,
          following: true,
          starred: false,
        }}
        location="/datasets/ds001/versions/1.0.0"
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
