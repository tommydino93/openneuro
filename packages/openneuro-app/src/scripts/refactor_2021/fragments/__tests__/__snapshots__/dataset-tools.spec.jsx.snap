// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetTools component renders tools for public datasets 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`DatasetTools component renders tools for public snapshots 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`DatasetTools component renders tools with common props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
